From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sotr <i@omc.hk>
Date: Wed, 15 Apr 2020 04:16:44 +0700
Subject: [PATCH] Remove stream for ender teleport


diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index 053233005086ae36f1091434f3f74a13ee83ef78..aa0c314e4c8ecc9bb0210b09f0db95bdccf1748f 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -3015,7 +3015,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
             WorldServer worldserver = (WorldServer) this.world;
 
             this.setPositionRotation(d0, d1, d2, this.yaw, this.pitch);
-            this.co().forEach((entity) -> {
+            this.collectPassengers().forEach((entity) -> { // Akarin - remove stream
                 worldserver.chunkCheck(entity);
                 entity.aE = true;
                 Iterator iterator = entity.passengers.iterator();
@@ -3262,6 +3262,14 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
     public Stream<Entity> co() {
         return Stream.concat(Stream.of(this), this.passengers.stream().flatMap(Entity::co));
     }
+    // Akarin start - non-stream version
+    public List<Entity> collectPassengers() {
+      List<Entity> result = Lists.newArrayList(this);
+      for (Entity passenger : this.passengers)
+        result.addAll(passenger.collectPassengers());
+      return result;
+    }
+    // Akarin end
 
     public boolean hasSinglePlayerPassenger() {
         Set<Entity> set = Sets.newHashSet();
